src/app/accounts/register.component.ts

```diff
@@ -96,6 +99,12 @@ export class RegisterComponent extends BaseRegisterComponent {
         if (this.policies != null) {
             this.enforcedPolicyOptions = await this.policyService.getMasterPasswordPolicyOptions(this.policies);
         }
+
+        const referenceId = ('; ' + document.cookie).split('; reference=').pop().split(';').shift();
+        this.referenceId = this.referenceId || referenceId;
+        if (this.referenceId === '') {
+            this.referenceId = null;
+        }
```

Comment:
Does this need to be moved into the queryParams subscribe callback? That is async

Reply:
Yes it does, and this has now been moved, verified after some testing that even with no query parameters this callback is fired from the queryParams observable, before ngOnInit completes so I bet the observable fires immediately but is still async, moved code to prevent a race condition.